README.rst

```diff
@@ -96,6 +96,70 @@ Submitting Pull Requests
+
+In general, we want to keep our fork up-to-date with changes from the `upstream gosnowflake repo <https://github.com/snowflakedb/gosnowflake>`_. Follow these steps to pull upstream changes into our fork.
+
+First, add the upstream repo as a remote:
+
+.. code-block:: bash
+
+    git remote add upstream git@github.com:snowflakedb/gosnowflake.git
+    git fetch upstream master
+
+To see the number of commits in ``upstream/master`` but not in ``origin/master``:
+
+.. code-block:: bash
+
+    git log --oneline origin/master..upstream/master | wc -l
+
+To see the number of commits in ``origin/master`` but not in ``upstream/master``:
+
+.. code-block:: bash
+
+    git log --oneline upstream/master..origin/master | wc -l
+
+Next, we want to create a merge commit that incorporates changes from the upstream repo with changes in our fork. These steps are based on `a blog post about maintaining long-lived forks <https://die-antwort.eu/techblog/2016-08-git-tricks-for-maintaining-a-long-lived-fork/>`_.
+
+.. code-block:: bash
+
+    # Create a branch reset-to-upstream that is up-to-date with upstream.
+    git checkout -b reset-to-upstream upstream/master
+
+    # Merge master into this branch, effectively ignoring all changes from master.
+    git merge --strategy=ours master
+
+    # Switch back to master and merge the temporary branch into a new branch
+    # off of master (will be a fast-forward merge).
+    git checkout master
+    git checkout -b update-master-with-upstream
+    git merge reset-to-upstream
+
+    # List commits that only exist in the fork.
+    git log upstream/master..master --no-merges | cat
+
+    # Cherry-pick changes from the fork as needed.
+    git cherry-pick <hash> <hash> <hash>
+
+Now we can push the branch ``update-master-with-upstream`` to our fork's origin, and create a pull request.
+
+Once that's done, we can run the Multiplex benchmarks with the new version of gosnowflake. In your local copy of the mono-go repo, update Multiplex to use the version of gosnowflake in ``update-master-with-upstream``:
+
+.. code-block:: bash
+
+    go get github.com/sigmacomputing/gosnowflake@<new gosnowflake commit hash>
+
+Create a commit with this update, along with any fixes to get Multiplex to work with the latest driver version.
+
+To run the benchmark, populate the file ``multiplex/scripts/parameters.json`` with Snowflake testing credentials, which can be found in the Sigma Engineering 1Password vault. Then, run the benchmark script to compare the old and new branches, and redirect output to a csv file.
+
+.. code-block:: bash
+
+    ./multiplex/scripts/query_bench.sh <old commit hash> <new commit hash> > ~/tmp/gosnowflake-bench-2021-06-11.csv
+
+Once that's done running, add a new page to the `Snowflake Go Driver Updates workbook <https://staging.sigmacomputing.io/sigma-on-sigma/workbook/Snowflake-Go-Driver-Updates-Hm0VmWLJZbgi9m4hWayjK>`_ with the results from your benchmark run.
+
```

Comment:
We didn't get to these steps, but the next parts are:

1. If there were errors while running any of the queries or if there was a substantial regression in query runtime, investigate before merging
2. If everything looks good (all queries ran, median and p90 runtime are within a few percent), merge the commit into gosnowflake
3. Go back to Multiplex and `go get github.com/sigmacomputing/gosnowflake@master` so that we're still on the master branch
4. Get feedback on the Multiplex PR + merge